backend/app/usecase/url/urlupdater_test.go

```diff
@@ -0,0 +1,142 @@
+// +build !integration all
+
+package url
+
+import (
+	"testing"
+	"time"
+
+	"github.com/short-d/app/fw/assert"
+	"github.com/short-d/app/fw/timer"
+	"github.com/short-d/short/backend/app/entity"
+	"github.com/short-d/short/backend/app/usecase/keygen"
+	"github.com/short-d/short/backend/app/usecase/repository"
+	"github.com/short-d/short/backend/app/usecase/risk"
+	"github.com/short-d/short/backend/app/usecase/validator"
+)
+
+func TestURLUpdaterPersist_UpdateURL(t *testing.T) {
+	t.Parallel()
+
+	now := time.Now().UTC()
+	alias := "boGp9w35"
+	validNewAlias := "eBJRJJty"
+	validNewLongLink := "https://httpbin.org/get?p1=v1"
+
+	testCases := []struct {
+		name          string
+		alias         *string
+		availableKeys []keygen.Key
+		urls          urlMap
+		user          entity.User
+		urlUpdate     entity.URL
+		relationUsers []entity.User
+		relationURLs  []entity.URL
+		expHasErr     bool
+		expectedURL   entity.URL
+	}{
+		{
+			name:  "successfully update existing long link",
+			alias: &alias,
+			urls: urlMap{
+				"boGp9w35": entity.URL{
+					Alias:       "boGp9w35",
+					OriginalURL: "https://httpbin.org",
+				},
+			},
+			user: entity.User{
+				Email: "gopher@golang.org",
+			},
+			urlUpdate: entity.URL{
+				OriginalURL: validNewLongLink,
+			},
+			expHasErr: false,
+			expectedURL: entity.URL{
+				Alias:       "boGp9w35",
+				OriginalURL: validNewLongLink,
+			},
+		},
+		{
+			name:  "alias doesn't exist",
+			alias: &validNewAlias,
+			urls: urlMap{
+				"boGp9w35zzzz": entity.URL{
+					Alias:       "boGp9w35zzzz",
+					OriginalURL: "https://httpbin.org",
+				},
+			},
+			user: entity.User{
+				Email: "gopher@golang.org",
+			},
+			urlUpdate: entity.URL{
+				OriginalURL: validNewLongLink,
+			},
+			expHasErr:   true,
+			expectedURL: entity.URL{},
+		},
+		{
+			name:  "long link is invalid",
+			alias: &alias,
+			urls: urlMap{
+				"boGp9w35": entity.URL{
+					Alias:       "boGp9w35",
+					OriginalURL: "https://httpbin.org",
+				},
+			},
+			user: entity.User{
+				Email: "gopher@golang.org",
+			},
+			urlUpdate: entity.URL{
+				OriginalURL: "aaaaaaaaaaaaaaaaaaa",
+			},
+			expHasErr:   true,
+			expectedURL: entity.URL{},
+		},
```

Comment:
What happens if the user does NOT own the short link? Let's add a test case for that.

Reply:
Also for a case for malicious long link

Reply:
> What happens if the user does NOT own the short link? Let's add a test case for that.

For this test case is it okay if I include it outside of the `testCases` slice?

Reply:
Actually I figured out how to include it sorry!